### DISCUSSION TITLE
Custom Report Layouts

### DISCUSSION THREAD

**Discussion by chamusgilbert-ov** [microsoft/alguidelines#164] [BC Best Practices]
Hello,

I'm currently on version 14 BC On-Prem. Is there any thoughts on best practice regarding Custom Report Layouts vs. Modifying RDLC files in the code? Just wondering pros and cons. We're kind of mixed, doing both.

**Comment by pranav4186** [comment]
If it is just about modifying layouts in BC14 (or any BC/NAV Version where custom report layout feature is available), then it is recommended to use custom report layouts rather than modifying default RDLC layout.
If you are using newer version of BC where report extension is available then you should use that before copying standard report and creating new report.

Here are some Pros/Cons of Custom Report Layouts:
Pros:
 -you would not need to worry about changing report layouts while upgrading.
 -you can create multiple layouts and can keep track of changes.
-It is kind of an extension for the report layout so less worry about breaking standard layout.

Cons:
-If by mistake user changes layout in Report Layout Selection then it will be a problem! but this can be resolved by training and permissions.

Thanks
Pranav
